src/dbsync/src/dbsync.cpp

```diff
@@ -170,12 +170,56 @@ int dbsync_set_table_max_rows(const DBSYNC_HANDLE      /*handle*/,
     return 0;
 }
 
-int dbsync_sync_row(const DBSYNC_HANDLE /*handle*/,
-                    const cJSON*        /*js_input*/,
-                    result_callback_t   /*callback*/)
+int dbsync_sync_row(const DBSYNC_HANDLE handle,
+                    const cJSON*        js_input,
+                    result_callback_t   callback)
 {
-    // Dummy function for now.
-    return 0;
+    auto ret_val { -1 };
+    std::string errorMessage;
+    if (!handle || !js_input || !callback)
+    {
+        errorMessage += "Invalid input parameters.";
+    }
+    else
+    {
+        try
+        {
+            nlohmann::json result;
+            const auto callbackWrapper
+            {
+                [&result](ReturnTypeCallback resultType, const nlohmann::json& jsonResult)
+                {
+                    static std::map<ReturnTypeCallback, std::string> s_opMap
```

Comment:
you need to copy or reuse the other callbackWrapper that actually calls the callback captured in the capture list of the lambda =).

Reply:
Done. :)